src/main/java/org/reactome/release/resourcechecker/FTPFileResourceChecker.java

```diff
@@ -0,0 +1,203 @@
+
+	@Override
+	public void saveFileContents(Path downloadDestination) throws IOException {
+		FTPClient ftpClient = connectToFTPClient();
+
+		Files.deleteIfExists(downloadDestination);
+		ftpClient.retrieveFile(
+			getFtpFilePath(),
+			new FileOutputStream(downloadDestination.toFile())
+		);
+
+		logoutAndDisconnect(ftpClient);
+	}
+
+	/**
+	 * Returns the size of the FTP file being checked in bytes (0 if the file does not exist)
+	 *
+	 * @return Returns the FTP file's size in bytes
+	 */
+	@Override
+	public long getFileSize() {
+		return getFtpFile().getSize();
+	}
+
+	/**
+	 * Returns the name of the host server on which the FTP file being checked resides (e.g. ftp.uniprot.org)
+	 *
+	 * @return Name of the FTP host server
+	 */
+	public String getFtpServer() {
+		return this.getResourceURL().getHost();
+	}
+
+	/**
+	 * Returns the path to the FTP file on its host server as a String
+	 *
+	 * @return Remote path of the FTP file as a String
+	 */
+	public String getFtpFilePath() {
+		return this.getResourceURL().getPath();
+	}
+
+	/**
+	 * Returns the user name used to connect to the FTP server on which the file being checked resides (Defaults to
+	 * "anonymous" if not specified when the instance, on which this method is called, was constructed
+	 *
+	 * @return User name to connect to FTP server
+	 *
+	 * @see #getPassword()
+	 */
+	public String getUserName() {
+		return this.userName;
+	}
+
+	/**
+	 * Returns the password used to connect to the FTP server on which the file being checked resides (Defaults to
+	 * "help@reactome.org" if not specified (i.e. using the "anonymous" user name) when the instance, on which this
+	 * method is called, was constructed.
+	 *
+	 * NOTE: An e-mail address is used as the password when connecting anonymously to an FTP server as a courtesy
+	 * to the FTP server's site operators so they know who is accessing their service
+	 * (see https://stackoverflow.com/a/20031581)
+	 *
+	 * @return User name to connect to FTP server
+	 *
+	 * @see #getUserName()
+	 */
+	public String getPassword() {
+		return this.password;
+	}
+
+	private FTPFile getFtpFile() {
+		if (this.ftpFile == null) {
+			try {
+				this.ftpFile = retrieveFtpFile();
+			} catch (IOException e) {
+				logger.error("Unable to retrieve file " + getFtpFilePath() + " from FTP Server " + getFtpServer(), e);
+			}
+		}
+
+		return this.ftpFile;
+	}
+
+	private FTPFile retrieveFtpFile() throws IOException {
+		FTPClient ftpClient = connectToFTPClient();
+
+		List<FTPFile> ftpFiles = Arrays.asList(
+			ftpClient.listFiles(getFtpFilePath())
+		);
+
+		logoutAndDisconnect(ftpClient);
+
+		if (ftpFiles.size() == 1) {
+			return ftpFiles.get(0);
+		} else {
+			logger.error(getErrorMessageForNonUniqueFile(ftpFiles, getFtpFilePath()));
+			return new FTPFile();
+		}
+	}
+
+	private FTPClient connectToFTPClient() throws IOException {
+		FTPClient ftpClient = new FTPClient();
+
+		ftpClient.connect(getFtpServer());
+		ftpClient.enterLocalPassiveMode();
```

Comment:
Might need an option to toggle this (passive/active mode)  for specific resources. It's been a problem in the past, hasn't it?

Reply:
Yes - I'll add it in if any of the resources accessed by FTP won't work without active mode during testing.

Reply:
Created issue #5